src/pages/en/guides/deploy/github.md

```diff
@@ -73,9 +73,27 @@ You can deploy an Astro site to GitHub Pages by using [GitHub Actions](https://g
     :::tip
     See [the GitHub Pages Action documentation](https://github.com/marketplace/actions/github-pages-action) for different ways you can configure the final “Deploy to GitHub Pages” step.
     :::
+3. (Optionally) Set up a custom domain by adding the following `./public/CNAME` file to your project: 
 
-3. Commit the new workflow file and push it to GitHub.
-4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.
-5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.
+    ```txt title="public/CNAME"
+    sub.mydomain.com
+    ```
+
+    `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     
+
+4. Commit the new workflow file and push it to GitHub.  
+
+5. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.  
+
+6. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.  
+  
+
+7. (Optionally) Set up a custom domain by adding the following `./public/CNAME` file to your project:   
+
+
+    ```txt title="public/CNAME"
+    sub.mydomain.com
+    ```
 
+    `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     
```

Comment:
```suggestion
3. Commit the new workflow file and push it to GitHub.  

4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.  

5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.  
  

:::tip
If you want a custom domain rather than `user.github.io`, follow [the official Github Guide for configuring domains](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain) and make sure to place the `CNAME` file into your `public` folder.
:::
```